### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by benjp009**
# 👋 Micronaut Reviewer 

[React capstone project - Metrics webapp](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md)

Today, I am presenting to you the project of React Capstone : 

##### Home Page 
- [x] When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
- [x] Along with the category name, you will display some numeric values per category that come from the API.
- [x] When the user clicks (or taps) on a category item, the application navigates to the details page.

##### Details page
- [x] In the details page, the webapp retrieves data from the API to show detailed data in the category.
When the user clicks on the "Back" button (<), the user navigates to the home page.

#### Testing requirements
- [x] Create unit tests for pure functions in your Redux code.
- [x] Create integration tests for your application using the React Testing Library.
  - [x] You may need to mock the access to the API, so that your tests don't send actual requests.
  - [x] You may need to mock the connection to the Redux Store.

####  Technical requirements
- [x] The project is a single page application (SPA) built with React and Redux.
- [x] The data retrieved from the API should be stored in the Redux store.
- [x] You should filter the data that you retrieve from the API using a Filter stateless component.
- [x] Every page (the main page and the pages for each item) should have a unique route within the SPA.
- [x] The project should be deployed and accessible online (add a link to the online version of your app to your README file).

#### Project documentation
The link to the video is [here](https://www.loom.com/share/08be4509819b412ab436a0f349232b46) 

### Pair Programming 🎳

For this exercise, PP was not required.

Happy Review :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,101 @@
-# Getting Started with Create React App
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+# Covid19 Metrics
```
Comment (ZeenatLawal) [inline README.md:2]: - [x] The content of the video presentation of the project should include:
  - a description of the project.
  - a demo of the project features.
  - you should also highlight some interesting piece of code or something you built that you are very proud of.

The current video doesn't contain these, kindly record another one.

**Comment by benjp009** [comment]
Screenshot - 4 February 2022

File: README.md [hunk 2]
```diff
@@ -1,70 +1,101 @@
...
+## Visit And Open Files
 
-### Advanced Configuration
+[Visit Repo](https://github.com/benjp009/covid-tracker-react-capstone)
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+## Download Repo
 
-### Deployment
+[Download Repo](https://github.com/benjp009/covid-tracker-react-capstone/archive/refs/heads/main.zip)
```
Comment (ZeenatLawal) [inline README.md]: - [x] The project should be deployed and accessible online (add a link to the online version of your app to your README file), this requirement hasn't been met yet, kindly work on it.

File: src/reportWebVitals.js [hunk 3]
```diff
@@ -1,6 +1,12 @@
-const reportWebVitals = onPerfEntry => {
+const reportWebVitals = (onPerfEntry) => {
```
Comment (ZeenatLawal) [inline src/reportWebVitals.js]: - [x] All boilerplate files and icons should be deleted as they are not required for the project. e.g In the public folder(the logo images, manifest.json and robots.txt) and in the src folder(reportwebvitals).

File: src/components/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            </h2>
+            <h3 className="banner-h3">{globalData.TotalConfirmed}</h3>
+            <h4 className="banner-h4">{globalData.Date}</h4>
+          </div>
+        </div>
+        <div className="search-bar">
+          <h3 className="search-bar-h3">Search by country name</h3>
+          <input type="text" placeholder="Search" className="search-bar-input" />
+          <h4 className="search-bar-h4">STATS BY COUNTRY</h4>
+        </div>
```
Comment (ZeenatLawal) [inline src/components/HomePage.js]: - [x] The filter input is currently not working as the functionality hasn't been added yet. Kindly work on it to meet this requirement `When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the category name.`

File: src/styles/Footer.css [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/* style-for-footer */
+@import url('https://fonts.googleapis.com/css2?family=Lato&display=swap');
+
+.footer {
+  position: fixed;
+  bottom: 0;
+  width: 100%;
```
Comment (ZeenatLawal) [inline src/styles/Footer.css]: - [x] On the details page some parts of the data can't be seen fully because of the footer, kindly modify the styling.

**Review by ZeenatLawal — CHANGES_REQUESTED** [review]
Hi @benjp009, 

Good job so far!:clap: :clap:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎊 🎊 
- Gitflow is followed. ✔️
- PR has a title and summary. ✔️


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/reportWebVitals.js (hunk 3 above)
Comment (benjp009) [reply src/reportWebVitals.js]: done

File: src/styles/Footer.css (hunk 5 above)
Comment (benjp009) [reply src/styles/Footer.css]: fixed position removed

**Comment by benjp009** [comment]
All changes are done :)

File: README.md [hunk 6]
```diff
@@ -1,70 +1,105 @@
...
+- REACT
+- Redux
+- Webpack
+- Jest
 
-**Note: this is a one-way operation. Once you `eject`, you can't go back!**
+## Online live link
 
-If you aren't satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+[Visit project online](https://benjp009.github.io/covid-tracker-react-capstone/)
```
Comment (er-contreras) [inline README.md]: - [ ] As the last CR mentioned the readme file should have a live version of your project either with Heroku, Netlify, or Github pages. You are just redirecting to your readme file again. 
Screen Shot 2022-02-04 at 10 24 23

File: package.json [hunk 7]
```diff
@@ -1,14 +1,26 @@
...
+    "react-redux": "^7.2.6",
+    "react-router": "^6.2.1",
+    "react-router-dom": "^6.2.1",
     "react-scripts": "5.0.0",
+    "redux": "^4.1.2",
+    "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.4.1",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
```
Comment (er-contreras) [inline package.json:25]: - [ ] When I run npm start it open the http://localhost:3000/covid-tracker-react-capstone but the page just show a background color to pink. If you remove the 'covid-tracker-react-capstone' from the URL it works 😉

Screen Shot 2022-02-04 at 10 28 40

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @benjp009 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️

- [ ] The last CR addressed the next and still should be applied 

- The content of the video presentation of the project should include:
  - a description of the project.
  - a demo of the project features.
  - you should also highlight some interesting piece of code or something you built that you are very proud of.
The current video doesn't contain these, kindly record another one.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by benjp009** [comment]
# Micronaut Reviewer 

Page is deployed on [heroku](https://brave-keller-27adb6.netlify.app/) and not on **github pages**

File: README.md [hunk 8]
```diff
@@ -1,70 +1,105 @@
-# Getting Started with Create React App
```
Comment (SajjadAhmad14) [inline README.md:1]: The video link in the PR description has no content and the video is just a few seconds.

The content of the video presentation of the project should include:
  a description of the project.
  a demo of the project features.
  you should also highlight some interesting piece of code or something you built that you are very proud of.
  Please record a new video.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Benjamin,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 8 above)
Comment (benjp009) [reply README.md:1]: here is the [video](https://www.loom.com/share/9bd7dc065ee8469e9f89044dfa0007d1)

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :partying_face: 

Wooow👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by benjp009** [comment]
@Emmyn5600 thanks for your review :)

**Pull request merged**